sdk/communication/azure-communication-identity/azure/communication/identity/_shared/models.py

```diff
@@ -145,8 +152,24 @@ def __init__(self, value: str, **kwargs: Any) -> None:
         :keyword str raw_id: The raw ID of the identifier. If not specified, this will be constructed from
           the 'value' parameter.
         """
-        self.properties = PhoneNumberProperties(value=value)
+
         raw_id: Optional[str] = kwargs.get("raw_id")
+        asserted_id: Optional[str] = ""
+        is_anonymous: Optional[bool] = False
+
+        if raw_id is not None:
+            phone_number = raw_id[len(PHONE_NUMBER_PREFIX):]
+            is_anonymous = phone_number == PHONE_NUMBER_ANONYMOUS_SUFFIX
+            asserted_id_index = -1 if is_anonymous else phone_number.rfind("_") + 1
+            has_asserted_id = 0 < asserted_id_index < len(phone_number)
+            asserted_id = phone_number[asserted_id_index:] if has_asserted_id else ""
```

Comment:
here if there is not an asserted id we have to set it to None, as we have done for other platforms